test.js

```diff
@@ -158,6 +154,27 @@ describe('capture-exit', function() {
         exit.onExit(function () { });
       }).to.throw('Cannot install handler when exit is not captured.  Call `captureExit()` first');
     });
+
+    it('throws if an exit is already happening', function() {
```

Comment:
@stefanpenner yeah the problem is that our on exit running conflates two cases: one were async is possible (`beforeExit` or during an intercepted `process.exit`) and one where async is impossible (during `process.on('exit')`).

If we want to add support to this where we can, I think we'll need to actually treat the two cases separately but I would prefer we not do that until we actually have a use case for it.